### PR TITLE
PEP: module __getattr__

### PR DESCRIPTION
**Pull request by ilevkivskyi**
This is an alternative (or complementary?) proposal to PEP 549.

cc @gvanrossum

### CODE REVIEW

File: pep-9999.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
+PEP: 9999
```
Comment (gvanrossum) [inline pep-9999.rst]: I guess this can be PEP 562.

File: pep-9999.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  # main.py
+
+  from lib import old_function  # Works, but emits the warning
+
+There is a related proposal PEP 549 that proposes to support instance
+properties for a similar functionality. The difference is this PEP proposes
+a faster and simpler mechanism, but provides more basic customization.
+An additional motivation for this proposal is that PEP 484 already defines
+the use of module ``__getattr__`` for a this purpose in Python stub files,
```
Comment (JelleZijlstra) [inline pep-9999.rst]: remove "a"

File: pep-9999.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+This function will be called only if ``name`` is not found in the module
+through the normal attribute lookup.
+
+The reference implementation for this PEP can be found in [2]_.
+
+
+Backwards compatibility and impact on performance
+=================================================
+
+This PEP will break the code that uses module level (global) name
```
Comment (JelleZijlstra) [inline pep-9999.rst]: remove "the"

**Comment by ilevkivskyi** [comment]
@JelleZijlstra @gvanrossum Thanks, everything is fixed now.

**Comment by gvanrossum** [comment]
Thanks, I've merged it, but haven't had time to review carefully yet. Go ahead and post to python-ideas.

**Comment by gvanrossum** [comment]
Also, I renamed the file to pep-0562.rst (the convention is always 4 digits in the filename, though nowhere else).

**Comment by ilevkivskyi** [comment]
@gvanrossum I have no idea what happened, but there are now two files ``pep-562.rst`` and ``pep-0562.rst``, therefore the PEP appears twice on the website etc. When I pull from master, I don't see ``pep-562.rst``, so I can't make a PR removing it. Could you please remove the duplicate?

**Comment by gvanrossum** [comment]
I left the old one in by mistake, but I have since removed it from the repo
(
https://github.com/python/peps/commit/4f82ff5bf3805415658836d212b37911cf029dce).
I don't know why the index still has two copies, there's probably some
caching bug. I don't have access to the machine -- can you file a bug at
https://github.com/python/pythondotorg/issues ? (Someone should be able to
reset the cache, but they should also look into ensuring this doesn't
happen again.)

**Comment by ilevkivskyi** [comment]
I opened https://github.com/python/pythondotorg/issues/1147

**Comment by ethanhs** [comment]
For me it disappeared after the page updated for PEP 561.

**Comment by gvanrossum** [comment]
It's gone now.

**Pull request merged**